src/diffusers/models/transformers/transformer_kandinsky.py

```diff
@@ -0,0 +1,823 @@
+# Copyright 2025 The Kandinsky Team and The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch import BoolTensor, IntTensor, Tensor, nn
+from torch.nn.attention.flex_attention import (
+    BlockMask,
+    flex_attention,
+)
+
+from ...configuration_utils import ConfigMixin, register_to_config
+from ...loaders import FromOriginalModelMixin, PeftAdapterMixin
+from ...utils import (
+    USE_PEFT_BACKEND,
+    deprecate,
+    logging,
+    scale_lora_layers,
+    unscale_lora_layers,
+)
+from ...utils.torch_utils import maybe_allow_in_graph
+from ..attention import AttentionMixin, FeedForward
+from ..cache_utils import CacheMixin
+from ..embeddings import (
+    TimestepEmbedding,
+    get_1d_rotary_pos_embed,
+)
+from ..modeling_outputs import Transformer2DModelOutput
+from ..modeling_utils import ModelMixin
+from ..normalization import FP32LayerNorm
+
+logger = logging.get_logger(__name__)
+
+
+def exist(item):
+    return item is not None
+
+
+def freeze(model):
+    for p in model.parameters():
+        p.requires_grad = False
+    return model
+
+
+@torch.autocast(device_type="cuda", enabled=False)
```

Comment:
```suggestion
```
Can we remove the `@autocast` decorator here? It's not doing anything since we're explicitly creating the tensor in `float32`, no? but correct me if I'm wrong

in general we should make sure the pipeline runs as expected without autocast, let's try to explicitly manage the dtype instead